pkg/api/transport_errcatch.go

```diff
@@ -55,9 +62,29 @@ func (e *ErrCatchTransport) RoundTrip(req *http.Request) (*http.Response, error)
 
 	res, err := e.rt.RoundTrip(req)
 	if res != nil {
-		// nolint
-		httputil.DumpResponse(res, res.Body != nil)
+		_, _ = httputil.DumpResponse(res, res.Body != nil)
+
+		// When the content type is "text/html", a bit of tweaking is required
+		// for the response to be marshaled to  JSON. Using the standard error
+		// definition and populating it with parts of the request so the error
+		// can be identified.
+		if strings.Contains(res.Header.Get("Content-Type"), "text/html") {
+			res.Header.Set("Content-Type", "application/json")
+			res.Body = newProxyBody(req, res.StatusCode)
+		}
 	}
 
 	return res, err
 }
+
+func newProxyBody(req *http.Request, code int) io.ReadCloser {
+	return mock.NewStructBody(models.BasicFailedReply{
+		Errors: []*models.BasicFailedReplyElement{
+			{
+				Code:    ec.String(strconv.Itoa(code)),
+				Fields:  []string{req.Method + " " + req.URL.EscapedPath()},
```

Comment:
any reason why `fmt.Sprintf` is not used here? Mostly curious - not a real comment tbh

Reply:
I was lazy, I'll fix it